File: Examples/VisualBasic/Working-Shapes/ReadAllShapeProps.vb

Imports Aspose.Diagram
Imports System

Public Class ReadAllShapeProps
    Public Shared Sub Run()
        ' ExStart:ReadAllShapeProps
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_Shapes()

        ' call a Diagram class constructor to load the VSDX diagram
        Dim diagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsdx"))
        ' get page by name
        Dim page As Page = diagram.Pages.GetPage("Page-3")

        For Each shape As Aspose.Diagram.Shape In page.Shapes
            If shape.Name = "Process1" Then
                For Each [property] As Prop In shape.Props
                    Console.WriteLine([property].Label.Value + ": " + [property].Value.Val)
                Next
                Exit For
            End If
        Next
        ' ExEnd:ReadAllShapeProps
    End Sub
End Class

File: Examples/CSharp/Working-Shapes/ReadAllShapeProps.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Diagram.Examples.CSharp.Working_Shapes
{
    public class ReadAllShapeProps
    {
        public static void Run()
        {
            // ExStart:ReadAllShapeProps
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Shapes();

            // Call a Diagram class constructor to load the VSDX diagram
            Diagram diagram = new Diagram(dataDir + "Drawing1.vsdx");
            // Get page by name
            Page page = diagram.Pages.GetPage("Page-3");

            foreach (Aspose.Diagram.Shape shape in page.Shapes)
            {
                if (shape.Name == "Process1")
                {
                    foreach (Prop property in shape.Props)
                    {
                        Console.WriteLine(property.Label.Value + ": " + property.Value.Val);
                    }
                    break;
                }
            }
            // ExEnd:ReadAllShapeProps
        }
    }
}
